### PR TITLE
refactor(batch): Use futures-async-stream to implement OrderBy executor 

### PR DESCRIPTION
**Pull request by D2Lark**
## What's changed and what's your intention?

Implement OrderBy using Executor2 trait.

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)
close https://github.com/singularity-data/risingwave/issues/1940

### CODE REVIEW

**Comment by liurenjie1024** [comment]
bors try

**Comment by D2Lark** [comment]
I SimpleThere are two more significant changes

- I refer the risinglight implementation to simplify the logic and pass the unit tests
- I've removed the encode key for now, I don't see that variable being used anywhere

File: src/batch/src/executor2/order_by.rs [hunk 1]
```diff
@@ -63,186 +56,74 @@ impl BoxedExecutorBuilder for OrderByExecutor {
...
-            }
-            chunk_size += 1;
-            self.push_heap_for_chunk(top.chunk_idx);
-        }
-        if chunk_size == 0 {
-            return Ok(None);
+            .map(|row_ref| row_ref.to_owned_row())
+            .collect::<Vec<_>>();
+        let data_chunk = DataChunk::from_rows(&rows, &data_types)?;
+        let chunks = DataChunk::rechunk(&[Arc::new(data_chunk)], self.chunk_size)?;
```
Comment (liurenjie1024) [inline src/batch/src/executor2/order_by.rs:103]: These two lines will consume 3 times memory than original version.

**Review by liurenjie1024 — COMMENTED** [review]
Please keep original version. The encodings was used to vectorized comparison.

File: src/batch/src/executor2/order_by.rs (hunk 1 above)
Comment (D2Lark) [reply src/batch/src/executor2/order_by.rs:103]: > Please keep original version. The encodings was used to vectorized comparison.

ok, I will use the original version

**Comment by liurenjie1024** [comment]
Please resolve conflicts.

**Comment by D2Lark** [comment]
ready for review @liurenjie1024

**Comment by liurenjie1024** [comment]
> 

Thanks, will take a look.

**Review by liurenjie1024 — APPROVED** [review]
LGTM

**Comment by D2Lark** [comment]
if this pr is merged ,we can close https://github.com/singularity-data/risingwave/issues/1948 and https://github.com/singularity-data/risingwave/issues/1174

**Comment by liurenjie1024** [comment]
Merged in #2327

**Pull request closed without merging**